### PR TITLE
feat(create the docs): add docz to create the documentation

### PR DESCRIPTION
**Pull request by nicholasess**
Hi Everyone!

I created the documentation of react-native-share with [docz](https://www.docz.site/).

To see the example running in my fork -> [link](https://nicholasess.github.io/react-native-share/#/)

- [x] README.md
- [x] Github Pages
- [x] Images worked
- [x] Dev dependencies

I needed to close my pr https://github.com/react-native-community/react-native-share/pull/356, because I merged the branch docs in master to create the example

Thanks

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1 @@
+docs
```
Comment (jgcmarins) [inline .eslintignore]: nice, maybe add this to `.gitignore` and `.npmignore`

**Comment by jgcmarins** [comment]
do not remove README.md

File: package.json [hunk 2]
```diff
@@ -60,6 +62,10 @@
     "release:major": "node ./changelog --major && npm version major && git push origin && git push origin --follow-tags",
     "release:minor": "node ./changelog --minor && npm version minor && git push origin && git push origin --follow-tags",
     "release:patch": "node ./changelog --patch && npm version patch && git push origin && git push origin --follow-tags",
-    "release:version": "node ./changelog --version $VERSION && git push origin && git push origin --follow-tags"
+    "release:version": "node ./changelog --version $VERSION && git push origin && git push origin --follow-tags",
+    "release:docz": "rm -rf docs && yarn docz build"
+  },
+  "dependencies": {
+    "react-dom": "^16.4.2"
```
Comment (jgcmarins) [inline package.json]: this should be a dev dep

**Comment by jgcmarins** [comment]
what is this 2mb file: https://github.com/react-native-community/react-native-share/blob/45458dea7a60e26bd0ecab606c67286a92bbfdf2/docs/static/js/vendors.a848727b.js
?

File: package.json (hunk 2 above)
Comment (nicholasess) [reply package.json]: Nice, I will change to dev

**Comment by nicholasess** [comment]
@jgcmarins I don’t know what is the 2mb, but I will update the .npmignore with docs folder and when had someone PR to documentation, you will need to rebuild

**Comment by yeomann** [comment]
i remember about Docz. it's great. i can not see the example. Link seems to be broken. @nicholasess

**Comment by nicholasess** [comment]
@yeomann I fixed the link

**Comment by nicholasess** [comment]
@jgcmarins the last commit has broken, but I just executed ```git pull``` and committed the files.

**Pull request closed without merging**